Shared/Features/Hymns/HymnView.swift

```diff
@@ -43,12 +43,12 @@ struct HymnView: View {
             }
         }
         .sheet(isPresented: $showCollectionModal) {
-            if let model = viewModel.model {
-                AddToCollectionView(hymnId: model.id, onDismiss: {
+                AddToCollectionView(hymnId: hymn.id, onDismiss: {
```

Comment:
We're always sending the initial `hymn` here.
So if we switch the hymnal we have to use the `viewModel.model`

Reply:
At this point `viewModel.model` is nil, so when the logic within this block is called the sheet is presented but nothing is displayed on it, that's why i opted to use the instance of hymn. 

Otherwise, another approach to solve the bug would be to redo the views and viewModel to always read from a single source of truth.